ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -51,6 +59,20 @@
    */
   public static final long Infinite_Time = -1;
 
+  /**
+   * Maximum supported time in millis
+   * @return the max supported time in millis
+   */
+  public static long getMaxSupportedTimeInMs() {
+    Date date = null;
+    try {
+      SimpleDateFormat sdf = new SimpleDateFormat(maxAllowableDateForEpochTimeFormat);
+      date = sdf.parse(maxAllowableDateForEpochTime);
+    } catch (ParseException e) {
```

Comment:
Catching like this isn't right, leads to null pointer exception later. Also, why do this dynamically when this value is a constant?

I actually think you should just make this a function of Integer.MAX_VALUE and . Why do we care about year 2038?